### PR TITLE
Add help view for keyboard shortcuts

### PR DESCRIPTION
**Pull request by Gomez**

### CODE REVIEW

File: js/templates/keyboard-shortcuts.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<tbody>
+<tr>
+	<th>Keyboard shortcut</th>
```
Comment (ChristophWurst) [inline js/templates/keyboard-shortcuts.html]: l10n? :wink:

File: js/templates/settings.html [hunk 2]
```diff
@@ -5,6 +5,8 @@
 	   class="button new-button"
 	   href="{{addAccountUrl}}">{{ t 'Add mail account' }}</a>
 
+	<p><a id="keyboard_shortcuts"
+	   href="{{keyboardShortcutUrl}}">{{ t 'Keyboard shortcuts' }}</a></p>
```
Comment (ChristophWurst) [inline js/templates/settings.html]: closing tag should be in the next line

File: js/views/appview.js [hunk 3]
```diff
@@ -159,6 +161,9 @@ define(function(require) {
 				}));
 			}
 		},
+		showKeyboardShortcuts: function() {
+			this.content.show(new KeyboardShortcutView({}));
```
Comment (ChristophWurst) [inline js/views/appview.js]: please also store which view is active (take a look at the other methods)

File: js/views/settings.js [hunk 4]
```diff
@@ -22,18 +22,24 @@ define(function(require) {
...
+				keyboardShortcutUrl: OC.generateUrl('apps/mail/#keyboardShortcut')
 			};
 		},
 		regions: {
 			accountsList: '#settings-accounts'
 		},
 		events: {
-			'click #new_mail_account': 'addAccount'
+			'click #new_mail_account': 'addAccount',
+			'click #keyboard_shortcuts': 'showKeyboardShortcuts'
```
Comment (ChristophWurst) [inline js/views/settings.js]: no underscores please

**Comment by Gomez** [comment]
All tests pass, ready for review @nextcloud/mail

File: js/tests/views/settings_spec.js [hunk 5]
```diff
@@ -36,14 +36,6 @@ define(['views/settings', 'views/helper'], function(SettingsView) {
 		it('produces the correct HTML', function () {
 			settingsview.render();
 
-			html = settingsview.el.innerHTML.trim();
```
Comment (ChristophWurst) [inline js/tests/views/settings_spec.js]: Interesting approach to fix the failing tests 😜

**Comment by jancborchardt** [comment]
Working on the design atm :)

File: js/tests/views/settings_spec.js (hunk 5 above)
Comment (Gomez) [inline js/tests/views/settings_spec.js]: :) I know. HTML comparison is a pain. Lets check for the link text only. Should be good enough? Or?

File: js/tests/views/keyboardshortcuts_spec.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+		beforeEach(function () {
+			keyboardshortcutsview = new KeyboardShortcutsView({});
+		});
+
+		it('produces the correct HTML', function () {
+			keyboardshortcutsview.render();
+
+		});
+	});
+});
```
Comment (ChristophWurst) [inline js/tests/views/keyboardshortcuts_spec.js]: new line missing

**Comment by jancborchardt** [comment]
Added a commit to improve the design, please also review core https://github.com/nextcloud/server/pull/4183 (needs backports also)

**Comment by jancborchardt** [comment]
capture du 2017-03-30 12-51-15

**Comment by jancborchardt** [comment]
Please also review the backports for server stable11 https://github.com/nextcloud/server/pull/4257 and stable10 https://github.com/nextcloud/server/pull/4258 – until Monday please since there’s a freeze. :)

**Comment by ChristophWurst** [comment]
Fixed/added proper URL routing support.

**Comment by ChristophWurst** [comment]
@nextcloud/mail please give this a quick test. This is ready to be integrated into master IMO.

**Comment by ChristophWurst** [comment]
The major changes are from @Gomez which I've approved -> merging this now 🚀

**Pull request merged**